tasks/task_database_cue.c

```diff
@@ -54,16 +55,192 @@ struct magic_entry
+   { 0x009320,   "Sony - PlayStation",            "\x50\x4c\x41\x59\x53\x54\x41\x54\x49\x4f\x4e",                   11},
+   { 0,          NULL,                            NULL,                                                             0}
 };
 
+/**
+ * Get the total number of occurrences of a character in the given string.
+ */
+static int count_occurrences_single_character(char *str, char t)
+{
+   int ctr = 0;
+   int index = -1;
+   int i;
+
+   for (i = 0; str[i] != '\0'; ++i) {
+      if (t == str[i])
+         ++ctr;
+   }
+
+   index = ctr;
+   return index;
+}
+
+/**
+ * Replaces all spaces with the given character.
+ */
+static void replace_space_with_single_character(char *str, char t)
+{
+   int new_char;
+   int ctr = 0;
+   char *dest = str;
+
+   while (str[ctr])
+   {
+      new_char = str[ctr];
+      if (isspace(new_char))
+         new_char = t;
+      *dest++ = new_char;
+      ctr++;
+   }
+
+   *dest = '\0';
+}
+
+/**
+ * Replaces multiple spaces with a single space in a string.
+ */
+static void replace_multi_space_with_single_space(char *str)
+{
+   char *dest = str;
+
+   while (*str != '\0')
+   {
+      while (*str == ' ' && *(str + 1) == ' ')
+      str++;
+
+      *dest++ = *str++;
+   }
+
+   *dest = '\0';
+}
+
+/**
+ * Removes leading and trailing spaces and tabs from the given string.
+ */
+static void left_and_right_trim_spaces(char *s)
+{
+   int  i,j;
+
+   for(i=0;s[i]==' '||s[i]=='\t';i++);
+
+   for(j=0;s[i];i++)
+   {
+      s[j++]=s[i];
+   }
+   s[j]='\0';
+   for(i=0;s[i]!='\0';i++)
+   {
+      if(s[i]!=' '&& s[i]!='\t')
+         j=i;
+   }
+   s[j+1]='\0';
+}
+
+/**
+ * Remove all spaces from the given string.
+ */
+static void remove_spaces (char* str_trimmed, const char* str_untrimmed)
+{
+   while (*str_untrimmed != '\0')
+   {
+      if(!isspace(*str_untrimmed))
+      {
+         *str_trimmed = *str_untrimmed;
+         str_trimmed++;
+      }
+      str_untrimmed++;
+   }
+   *str_trimmed = '\0';
+}
+
+/**
+ * Retrieve the last occurance of the given character in a string.
+ */
+static int index_last_occurance(char str[], char t)
+{
+   const char * ret = strrchr(str, t);
+   if (ret)
+      return ret-str;
+
+   return -1;
+}
+
+/**
+ * Find the position of a substring in a string.
+ */
+static int find_index_substring_string(const char* str1, const char* str2)
+{
+   int index;
+
+   int dlenstr = strlen(str1);
+   if (dlenstr > 0)
+   {
+      const char *pfound = strstr(str1, str2);
+      if (pfound != NULL)
+      {
+         index = (pfound - str1);
+         return index;
+      }
+   }
+
+   return -1;
+}
+
+/**
+ * Given a filename and position, find the associated disc number.
+ */
+static int find_disc_number(const char* str1, int index)
+{
+   char disc;
+   int disc_number = 0;
+
+   disc = str1[index + 6];
+   disc_number = disc - '0';
+
+   if (disc_number >= 1)
```

Comment:
What is disks are alphanumeric? "(Disk A)" should this also check `disc_number <= 9`?

Reply:
Thanks for this review.  I will make these code changes as they are well written.  Should get around to it tonight.

Reply:
I put in a switch/case for Aa-Ii